src/wp-includes/block-bindings/sources/post-meta.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Add the post_meta source to the Block Bindings API.
+ *
+ * @since 6.5.0
+ * @package WordPress
+ */
+function post_meta_source_callback( $source_attrs ) {
+	// Use the postId attribute if available
+	if ( isset( $source_attrs['postId'] ) ) {
+		$post_id = $source_attrs['postId'];
+	} else {
+		// $block_instance->context['postId'] is not available in the Image block.
+		$post_id = get_the_ID();
+	}
+
+	// If a post isn't public, we need to prevent
+	// unauthorized users from accessing the post meta.
+	$post = get_post( $post_id );
+	if ( ( $post && 'publish' !== $post->post_status && ! current_user_can( 'read_post', $post_id ) ) || post_password_required( $post_id ) ) {
```

Comment:
@artemiomorales @SantosGuillamot @gziolo I'm thinking this needs a little further tweaking.

* `is_post_publicly_viewable( $post )` could be used if we want to check if the post status and post type are public, not required but this is generally how the Loop Query block logic works (but it uses WP_Query stuff)
* or at minimal, instead of using a manual 'publish' check here -- I would suggest using the more flexible `is_post_status_viewable( $post->post_status )` call here would also allow for filtering

Reply:
Thanks @sc0ttkclark !

`is_post_publicly_viewable()` uses `is_post_status_viewable()` under the hood so I've used the former following your tip in a1e1bac8b3c9dfa0bc0a2be644c3b4382a2e3dc3

Reply:
@michalczaplinski Yeah I wasn't sure which direction you wanted to go, glad it was something more robust.

As for the current condition here from your commit, it needs further tweaking as the logic is not quite right yet. It's the right function calls but I think the `is_post_publicly_viewable()` check needs to be negated like:

```
	if ( ( ! is_post_publicly_viewable( $post ) && ! current_user_can( 'read_post', $post_id ) ) || post_password_required( $post ) ) {
```

Reply:
Oh, you're right. Thanks for pointing that out. done in d189f50862512d8e0d7127e01fbee433a246abff

Reply:
`post_password_required` could take the same `$post` object instead of `$post_id` at this point too if we want to slightly optimize that call.

Reply:
Thanks for the tip! Optimized the call to `post_password_required` in https://github.com/WordPress/wordpress-develop/pull/5888/commits/d7201cea4e859f6830199544e72a20f07b72537e.